### PR TITLE
Fix some RBIs that actually take blocks

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Pre-work for #4061

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_J4cm2vYDACdsHs
→ https://go/builds/bui_J4cmDfj6tNxdsu

File: rbi/core/enumerable.rbi [hunk 1]
```diff
@@ -844,16 +844,16 @@ module Enumerable
...
-    .returns(Elem)
+    .returns(T.untyped)
   end
   sig do
     params(
-        blk: T.proc.params(arg0: Elem, arg1: Elem).returns(Elem),
+        blk: T.proc.params(arg0: T.untyped, arg1: Elem).returns(T.untyped),
     )
-    .returns(T.nilable(Elem))
+    .returns(T.untyped)
```
Comment (elliottt) [inline rbi/core/enumerable.rbi:856]: Why switch these to `T.untyped`?

**Review by elliottt — APPROVED** [review]
Looks good!

Why did so many signatures get `Elem` removed in favor of `T.untyped`?

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_J5WctwcuJixeM7
→ https://go/builds/bui_J5WcFQHGrB8Ot6

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_J5o0eLM0LxhwfQ
→ https://go/builds/bui_J5o0lPx3H4b9Lm

**Comment by jez** [comment]
> Why did so many signatures get `Elem` removed in favor of `T.untyped`?

The accumulator of reduce is not required to accumulate into the same type. You'll notice that for the non-block overloads, they take `T.untyped`. Many have tried and failed to add a proper generic type for `reduce`, but Sorbet's generics aren't good enough for it.

**Pull request merged**